xbmc/utils/StringUtils.cpp

```diff
@@ -1032,6 +1036,51 @@ int64_t StringUtils::AlphaNumericCompare(const wchar_t* left, const wchar_t* rig
   wchar_t lc, rc;
   int64_t lnum, rnum;
   bool lsym, rsym;
+
+#ifdef TARGET_ANDROID
+  if (!g_langInfo.UseLocaleCollation())
+  {
+    UErrorCode ustatus = U_ZERO_ERROR;
+    try
+    {
+      auto ucoll = std::unique_ptr<UCollator, decltype(&ucol_close)>(
+          ucol_open(g_langInfo.GetISOLocale().c_str(), &ustatus), &ucol_close);
```

Comment:
Isn't it quite wasteful to call `ucol_open` for every string comparison?

Reply:
I think so. Shall I store the `UCollator` pointer in `g_LangInfo`?

Reply:
Hm I'm not sure it belongs there since (for now) it's platform specific, so it would need to be a new service. The better solution could be to keep some state in the function that does the actual sorting. If you haven't noticed a performance problem, let's keep it as-is for now (I just noticed that the same applies to `std::collate` which is used in the non-android code path)